x/mongo/driver/topology/topology_options_test.go

```diff
@@ -85,6 +90,74 @@ func TestLoadBalancedFromConnString(t *testing.T) {
 	}
 }
 
+type testAuthenticator struct{}
```

Comment:
Can add a compile check asserting that this type implements `driver.Authenticator`?